src/sentry/tasks/integrations/github/pr_comment.py

```diff
@@ -318,3 +325,43 @@ def github_comment_reactions():
             continue
 
         metrics.incr("github_pr_comment.comment_reactions.success")
+
+
+# TODO(adas): Change the client typing to allow for multiple SCM Integrations
+def safe_for_comment(
+    gh_client: GitHubAppsClient, repository: Repository, pull_request: PullRequest
+) -> bool:
+    try:
+        pullrequest_resp = gh_client.get_pullrequest(
+            repo=repository.name, pull_number=pull_request.key
+        )
+    except ApiError as e:
+        if e.json and RATE_LIMITED_MESSAGE in e.json.get("message", ""):
+            metrics.incr(
+                OPEN_PR_METRIC_BASE.format(key="api_error"),
+                tags={"type": "gh_rate_limited", "code": e.code},
+            )
+        elif e.code == 404:
+            metrics.incr(
+                OPEN_PR_METRIC_BASE.format(key="api_error"),
+                tags={"type": "missing_gh_pull_request", "code": e.code},
```

Comment:
when we're trying to find PR associated with the merge commit for merged PR comments, we don't error if we can't find a single associated pull request. should we be erroring here or returning?

https://github.com/getsentry/sentry/blob/332fdb7d0f9c02bdd64dbad9d3301af139a40336/src/sentry/tasks/commit_context.py#L64-L75

Reply:
Yeah, we probably should, cause this line will fail otherwise:
https://github.com/getsentry/sentry/blob/332fdb7d0f9c02bdd64dbad9d3301af139a40336/src/sentry/tasks/commit_context.py#L77

Reply:
erroring or returning?

Reply:
log an error and return. We don't want to cause the rest of commit context to fail cause of a missing PR

Reply:
right okay